### DISCUSSION TITLE
How can drag and select Link ?

### DISCUSSION THREAD

**Discussion by reshin-pulapaddy** [clientIO/joint#2294] [Q&A]
### Introduction

https://codepen.io/jointjs/pen/LYBKgPN/2a8a09292272afb06798d9a5d53fecf8
[2024](https://github.com/clientIO/joint/discussions/2024)

I am using the above example for implementing drag and select all elements including Link, but when I zoom in or zoom out the canvas it is not working properly.
![joint js](https://github.com/clientIO/joint/assets/117301232/020b4dd6-da4c-4c80-9951-b26f0fea207e)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
You need to append the selection lasso (`rect`) to the `paper.viewport` instead of `paper.svg`.

```js
rect.appendTo(paper.viewport);
```

A similar issue: https://github.com/clientIO/joint/issues/2280#issuecomment-1663610684

**Reply by reshin-pulapaddy** [reply]
Thank you @kumilingus
